solidity/random-beacon/contracts/DKGValidator.sol

```diff
@@ -0,0 +1,230 @@
+    function validateFields(DKG.Result calldata result)
+        public
+        view
+        returns (bool isValid, string memory errorMsg)
+    {
+        // Group public key needs to be 128 bytes long.
+        if (result.groupPubKey.length != 128) {
+            return (false, "Malformed group public key");
+        }
+
+        // The number of misbehaved members can not exceed the threshold.
+        // Misbehaved member indices needs to be unique, between [1,64],
+        // and sorted in ascending order.
+        uint8[] calldata misbehavedMembersIndices = result
+            .misbehavedMembersIndices;
+        if (groupSize - misbehavedMembersIndices.length < activeThreshold) {
+            return (false, "Too many members misbehaving during DKG");
+        }
+        if (misbehavedMembersIndices.length > 1) {
+            if (
+                misbehavedMembersIndices[0] < 1 ||
+                misbehavedMembersIndices[misbehavedMembersIndices.length - 1] >
+                64
+            ) {
+                return (false, "Corrupted misbehaved members indices");
+            }
+            for (uint256 i = 1; i < misbehavedMembersIndices.length; i++) {
+                if (
+                    misbehavedMembersIndices[i - 1] >=
+                    misbehavedMembersIndices[i]
+                ) {
+                    return (false, "Corrupted misbehaved members indices");
+                }
+            }
+            uint8 last = misbehavedMembersIndices[
+                misbehavedMembersIndices.length - 1
+            ];
+            if (last < 1 || last > groupSize) {
+                return (false, "Corrupted misbehaved members indices");
+            }
+        }
+
+        // Each signature needs to be 65 bytes long and signatures need to be
+        // provided.
+        uint256 signaturesCount = result.signatures.length / 65;
+        if (result.signatures.length == 0) {
+            return (false, "No signatures provided");
+        }
+        if (result.signatures.length % 65 != 0) {
+            return (false, "Malformed signatures array");
+        }
+
+        // We expect the same amount of signatures as the number of declared
+        // group member indices that signed the result.
+        uint256[] calldata signingMembersIndices = result.signingMembersIndices;
+        if (signaturesCount != signingMembersIndices.length) {
+            return (false, "Unexpected signatures count");
+        }
+        if (signaturesCount < groupThreshold) {
+            return (false, "Too few signatures");
+        }
+        if (signaturesCount > groupSize) {
+            return (false, "Too many signatures");
+        }
+
+        // Signing member indices needs to be unique, between [1,64], and sorted
+        // in ascending order.
+        if (
+            signingMembersIndices[0] < 1 ||
+            signingMembersIndices[signingMembersIndices.length - 1] > 64
```

Comment:
Maybe `groupSize` instead of just `64`?

Reply:
https://github.com/keep-network/keep-core/pull/2734/commits/5bc95364647baabb9b6c7951805b9fb2681dcdfb